scripts/top-up-cycles-ledger-account

```diff
@@ -1,38 +1,93 @@
 
 	Note: The destination account will be topped up by the desired amount minus fees.
+
+	Note: If you have insufficient cycles on mainnet:
+	* Check that you have ICP:
+	  dfx ledger balance --ic
+	* If not, send ICP to this account:
+	  dfx ledger --ic account-id
+	* Convert some ICP to cycles
+	  dfx cycles --ic convert --icp 1
+	* See how many cycles you have now:
+	  dfx cycles --ic balance
 	EOF
 }
 
-[[ "${1:-}" != "--help" ]] || {
+# Source the clap.bash file ---------------------------------------------------
+source "$SOURCE_DIR/clap.bash"
+# Define options
+clap.define short=n long=network desc="dfx network to use" variable=NETWORK default="local"
+clap.define short=c long=cycles desc="Number of cycles" variable=CYCLES default="1T"
+clap.define short=t long=to desc="The destinantion principal or canister name" variable=ACCOUNT default="backend"
+# Source the output file ----------------------------------------------------------
+source "$(clap.build)"
+
+: Verifies that the account has been specified
+[[ "${ACCOUNT}" ]] || {
+  echo "ERROR: Please supply the destination principal or canister name."
   print_help
-  exit 0
 }
 
-account="$1"
-sum="$2"
-account="$(dfx canister id "$account")" # Converts names to canister IDs
-echo "Balance before: $(dfx canister call cycles_ledger icrc1_balance_of '(record { owner = principal "'"$account"'"; subaccount = null })')"
-dfx wallet send "$(dfx canister id cycles_depositor)" "$sum"
-dfx canister call cycles_depositor deposit '
-(
-  record {
-    to = record {
-    owner = principal "'"$account"'";
-      subaccount = null;
-    };
-    memo = null;
-    cycles = '"$sum"' : nat;
-  },
-)
-'
-echo "Balance after: $(dfx canister call cycles_ledger icrc1_balance_of '(record { owner = principal "'"$account"'"; subaccount = null })')"
+: Converts e.g. 1T cycles to a number:
+CYCLES="${CYCLES//k/K}"
+CYCLES="${CYCLES//m/M}"
+CYCLES="${CYCLES//g/G}"
+CYCLES="${CYCLES//t/T}"
+CYCLES="${CYCLES//K/_000}"
+CYCLES="${CYCLES//M/_000_000}"
+CYCLES="${CYCLES//G/_000_000_000}"
+CYCLES="${CYCLES//T/_000_000_000_000}"
+
+: Converts canister names to IDs:
+ACCOUNT="$(dfx canister id "$ACCOUNT" --network "$NETWORK")"
+
+top_up_local() {
+  : "Tops up a local or devenv cycles account using a cycles_depositor canister."
+  dfx wallet send "$(dfx canister id cycles_depositor)" "$cycles" --network "$NETWORK"
+  dfx canister call cycles_depositor deposit --network "$NETWORK" '
+	(
+	  record {
+	    to = record {
+	    owner = principal "'"$ACCOUNT"'";
+	      subaccount = null;
+	    };
+	    memo = null;
+	    cycles = '"$cycles"' : nat;
+	  },
+	)
+	'
+}
+top_up_mainnet() {
+  : "Tops up a cycles account on mainnet or similar, where cycles cannot be magicked out of nowhere."
+  dfx cycles transfer "$ACCOUNT" "$cycles" --network "$NETWORK"
+}
+is_test_env() {
+  : "Checks whether this is a test environment."
+  : TODO: Add support for devenv
+  [[ "$NETWORK" == "local" ]]
```

Comment:
shall we add `test_fe_` here too? if for example we are playing with those, do we want to top_up staging backend? I imagine yes, it is needed, but just to double-confirm

Reply:
For the purposes of this function, `test_fe_1` is on mainnet and mainnet is not a test environment.  It's a test canister on a production environment.  But thank you for asking, I'll update the comment to clarify this.

Reply:
Done.